man/fi_cq.3.md

```diff
@@ -382,6 +383,72 @@ or fi_cq_sreadfrom.  This may be used to wake-up a thread
 that is blocked waiting to read a completion operation.  The fi_cq_signal
 operation is only available if the CQ was configured with a wait object.
 
+# COMPLETION FIELDS
+
+The CQ entry data structures share many of the same fields.  The meanings
+of these fields are the same for all CQ entry structure formats.
+
+*op_context*
+: The operation context is the application specified context value that
+  was provided with an asynchronous operation.  The op_context field is
+  valid for all completions.
+  
+*flags*
+: This specifies flags associated with the completed operation.  The
+  _Completion Flags_ section below lists valid flag values.  Flags are
+  set for all relevant completions.
+
+*len*
+: This len field only applies to completed receive operations.  It indicates
+  the size of received message data -- i.e. the number of bytes of data that
+  was actually placed into the associated receive buffer.  If an endpoint has
+  been configured with the FI_MSG_PREFIX mode, the len also reflects the size
+  of the prefix buffer.
+
```

Comment:
Actually, I might be mistaken. I’ll look into it.

- Ben


From: Sean Hefty <notifications@github.com<mailto:notifications@github.com>>
Reply-To: ofiwg/libfabric <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Monday, July 6, 2015 at 2:10 PM
To: ofiwg/libfabric <libfabric@noreply.github.com<mailto:libfabric@noreply.github.com>>
Cc: Benjamin Turrubiates <bturrubi@cisco.com<mailto:bturrubi@cisco.com>>
Subject: Re: [libfabric] fabric: Define CQ entry fields (#1140)


In man/fi_cq.3.md<https://github.com/ofiwg/libfabric/pull/1140#discussion_r33982782>:

> +: The operation context is the application specified context value that
> +  was provided with an asynchronous operation.  The op_context field is
> +  valid for all completions.
> +
> +*flags*
> +: This specifies flags associated with the completed operation.  The
> +  _Completion Flags_ section below lists valid flag values.  Flags are
> +  set for all relevant completions.
> +
> +*len*
> +: This len field only applies to completed receive operations.  It indicates
> +  the size of received message data -- i.e. the number of bytes of data that
> +  was actually placed into the associated receive buffer.  If an endpoint has
> +  been configured with the FI_MSG_PREFIX mode, the len also reflects the size
> +  of the prefix buffer.
> +


The fi_send and fi_recv calls don't take a size that includes the prefix buffer, so should completion fields reflect that behavior?
I didn't define the len for sends, only for receives. I wasn't sure what usnic did, so I just defined something, figuring I had a 50/50 chance of getting it correct. :) I agree that the completion should match the send/recv calls. Are you sure that send/recv don't take the prefix size? That seems unfortunate, as it makes it impossible to use the prefix with an iov.

—
Reply to this email directly or view it on GitHub<https://github.com/ofiwg/libfabric/pull/1140/files#r33982782>.